code/modules/admin/verbs/commandreport.dm

```diff
@@ -0,0 +1,151 @@
+	set category = "Admin.Events"
+	set name = "Create Command Report"
+
+	if(!check_rights(R_ADMIN))
+		return
+
+	SSblackbox.record_feedback("tally", "admin_verb", 1, "Create Command Report") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+	var/datum/command_report_menu/tgui = new(usr)
+	tgui.ui_interact(usr)
+
+/// Datum for holding the TGUI window for command reports.
+/datum/command_report_menu
+	/// The mob using the UI.
+	var/mob/ui_user
+	/// The name of central command that will accompany our report
+	var/command_name = CENTCOM_PRESET
+	/// Whether we are using a custom name instead of a preset.
+	var/custom_name
+	/// The actual contents of the report we're going to send.
+	var/command_report_content
+	/// Whether the report's contents are announced.
+	var/announce_contents = TRUE
+	/// The sound that's going to accompany our message.
+	var/played_sound = DEFAULT_ANNOUNCEMENT_SOUND
+	/// A static list of preset names that can be chosen.
+	var/static/list/preset_names = list(CENTCOM_PRESET, SYNDICATE_PRESET, WIZARD_PRESET, CUSTOM_PRESET)
+
+/datum/command_report_menu/New(mob/user)
+	ui_user = user
+
+/datum/command_report_menu/ui_state(mob/user)
+	return GLOB.admin_state
+
+/datum/command_report_menu/ui_close()
+	qdel(src)
+
+/datum/command_report_menu/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "CommandReport")
+		ui.open()
+
+/datum/command_report_menu/ui_data(mob/user)
+	var/list/data = list()
+	data["command_name"] = command_name
+	data["custom_name"] = custom_name
+	data["command_report_content"] = command_report_content
+	data["announce_contents"] = announce_contents
+	data["played_sound"] = played_sound
+
+	return data
+
+/datum/command_report_menu/ui_static_data(mob/user)
+	var/list/data = list()
+	data["command_name_presets"] = preset_names
+	data["announcer_sounds"] = list(DEFAULT_ANNOUNCEMENT_SOUND) + GLOB.announcer_keys
+
+	return data
+
+/datum/command_report_menu/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if(.)
+		return
+
+	switch(action)
+		if("update_command_name")
+			if(params["updated_name"] == CUSTOM_PRESET)
+				custom_name = TRUE
+			else if (params["updated_name"] in preset_names)
+				custom_name = FALSE
+
+			command_name = params["updated_name"]
+			. = TRUE
+		if("update_report_contents")
+			command_report_content = params["updated_contents"]
+			. = TRUE
+		if("set_report_sound")
+			played_sound = params["picked_sound"]
+			. = TRUE
+		if("toggle_announce")
+			announce_contents = !announce_contents
+			. = TRUE
+		if("submit_report")
+			if(!command_name)
+				to_chat(ui_user, "<span class='danger'>You can't send a report with no command name.</span>")
+				return
+			if(!command_report_content)
+				to_chat(ui_user, "<span class='danger'>You can't send a report with no contents.</span>")
+				return
+			send_announcement()
+			. = TRUE
+
+	return
```

Comment:
This isn't necessary, return is implicit, though I think you should just use `return TRUE` in place of every `. = TRUE`.